src/main/java/org/opensearch/security/transport/SecurityInterceptor.java

```diff
@@ -127,6 +129,8 @@ public <T extends TransportResponse> void sendRequestDecorate(AsyncSender sender
         final String origCCSTransientMf = getThreadContext().getTransient(ConfigConstants.OPENDISTRO_SECURITY_MASKED_FIELD_CCS);
 
         final boolean isDebugEnabled = log.isDebugEnabled();
+        final boolean useJDKSerialization = connection.getVersion().before(Version.V_3_0_0);
```

Comment:
Is this change only going to be shipped in 3.0.0 or is this change going to be backported to 2.x?

Reply:
It's not intended to be backported.

Reply:
FYI - https://github.com/opensearch-project/security/issues/2780#issuecomment-1582935431

Reply:
It seems like the only time to use this setting is when running in mixed mode with older clusters, is there a way to specifically detect that state?  This would simplify logic about which mode to run in today, as well as provide a way to handle updates to the serialization format in the future.

Reply:
qq - When we backport to an earlier version of OS and release, would this change be a part of **a newer minor version**? If the answer is yes, then we can easily detect mixed cluster state as we'd have different values for max and min node versions.

Reply:
There are planned minor releases for 2.X released regularly [1]. There will be no minor release update for 1.X line, we are only releasing patches on 1.3.X, but this wouldn't prevent backward/forward compatibility changes from going into a patch release.

- [1] https://opensearch.org/releases.html#release-schedule

Reply:
@cwperks Is this thread good to resolve?

Reply:
@parasjain1 This is resolved. Thank you for the linked answer. Posting below to put the answer inline. 

> It's because of the complication with the rolling upgrade scenario. During rolling upgrade we'll have a mixed cluster with the old nodes understanding JDK serialization and the new ones custom. To establish communication b/w the two types of nodes we depend on the remote node version. Say we release this change with V3.0.0, the code changes assume that a node running OS version prior to V3.0.0 understands JDK serialization and will (de)serialize the headers using JDK serialization only.

> This will not work if we backport it to 2.x as the assumption made will no longer hold.

Reply:
I'm unclear how this will work during a rolling upgrade, this inspection is happening on the current node, not the destination nodes version (if that can be known), I believe this will cause issues.

Lets add a rolling upgrade test so that 2.X & 3.0 versions are tested to prove that works when the headers are transited between versions and are handled correctly that would confirm this works as expected.

Reply:
Yes. This is the correct understanding.

I've tested the same with active ingestion and search traffic on an index with basic DLS/FLS configured, during a rolling upgrade. There's one more round of testing that I can perform after a set of recent changes.

Reply:
@cwperks The comment pasted above is outdated. I had discussions with @peternied and posted this update -
https://github.com/opensearch-project/security/pull/2802#issuecomment-1589075858

Conclusion being, we'll be able to port it to 2.x with a change in version check.